src/approaches/grammar_search_invention_approach.py

```diff
@@ -667,6 +670,101 @@ def _evaluate_with_operators(self,
         return score
 
 
+@dataclass(frozen=True, eq=False, repr=False)
+class _ExpectedNodesScoreFunction(_OperatorLearningBasedScoreFunction):
+    """Score a predicate set by learning operators and planning in the training
+    tasks.
+
+    The score corresponds to expected number of nodes that would need to be
+    created before a low-level plan is found. This calculation requires
+    estimating the probability that each goal-reaching skeleton is refinable.
+    To estimate this, we assume a prior on how optimal the demonstrations are,
+    and say that if a skeleton is found by planning that is a different length
+    than the demos, then the likelihood of the predicates/operators goes down
```

Comment:
is one direction handled differently from the other?

Reply:
not right now. but I think we're almost always (maybe always always) going to be in the situation where the demo len is >= the plan len.